modular_skyrat/modules/delam_emergency_stop/code/scram.dm

```diff
@@ -211,55 +186,57 @@
-
-	// Let the gas work for a few seconds to cool the crystal. If it has damage beyond repair, heal it a bit
-	addtimer(CALLBACK(src, PROC_REF(prevent_explosion)), 9 SECONDS)
-
-/// Shatter the supermatter chamber windows
-/obj/structure/window/reinforced/plasma/proc/shatter_window()
-	visible_message(span_danger("[src] shatters in the freon fire!"))
-	explosion(src, SHATTER_DEVASTATION_RANGE, SHATTER_HEAVY_RANGE, SHATTER_LIGHT_RANGE, SHATTER_FLAME_RANGE, SHATTER_FLASH_RANGE)
-	qdel(src)
+	// Let the gas work for a few seconds to start the cooling process.
+	addtimer(CALLBACK(src, PROC_REF(aftermath)), 2 SECONDS)
 
 /// The valiant little machine falls apart, one time use only!
 /obj/machinery/atmospherics/components/unary/delam_scram/proc/goodbye_friends()
-
 	// good job buddy, sacrificing yourself for the greater good
-	playsound(src, 'sound/machines/compiler/compiler-failure.ogg', 100, FALSE, MACHINE_SOUND_RANGE, ignore_walls = TRUE, use_reverb = TRUE, falloff_distance = MACHINE_SOUND_FALLOFF_DISTANCE)
+	playsound(
+		source = src,
+		soundin = 'sound/machines/compiler/compiler-failure.ogg',
+		vol = 100,
+		vary = FALSE,
+		extrarange = 15,
+		ignore_walls = TRUE,
+		use_reverb = TRUE,
+		falloff_distance = 10,
+	)
 	visible_message(span_danger("[src] beeps a sorrowful melody and collapses into a pile of twisted metal and foam!"), blind_message = span_danger("[src] beeps a sorrowful melody!"))
 	deconstruct(FALSE)
 
 /// Drain the internal energy, if the crystal damage is above 100 we heal it a bit. Not much, but should be good to let them recover.
-/obj/machinery/atmospherics/components/unary/delam_scram/proc/prevent_explosion()
+/obj/machinery/atmospherics/components/unary/delam_scram/proc/aftermath()
 	var/obj/machinery/power/supermatter_crystal/engine/damaged_sm = my_sm?.resolve()
 	if(!damaged_sm)
 		return
 
 	damaged_sm.name = "partially delaminated supermatter crystal"
 	damaged_sm.desc = "This crystal has seen better days, the glow seems off and the shards look brittle. Central says it's still \"relatively safe.\" They'd never lie to us, right?"
-	damaged_sm.explosion_power = SM_DAMAGED_EXPLOSION_POWER // if you fuck up again, yeesh
-
-	if(damaged_sm.damage > SM_PREVENT_EXPLOSION_THRESHOLD)
-		damaged_sm.damage = SM_PREVENT_EXPLOSION_THRESHOLD
+	damaged_sm.explosion_power += 5 // if you fuck up again, yeesh
+	for(var/current_gas_mix as anything in damaged_sm.current_gas_behavior)
+		var/datum/sm_gas/gas_mix = damaged_sm.current_gas_behavior[current_gas_mix]
+		if(istype(gas_mix, /datum/sm_gas/freon))
+			gas_mix.heat_resistance += 0.4
+			continue
+		gas_mix.heat_modifier += 1.5
+		gas_mix.heat_resistance -= 0.4
 
-	damaged_sm.internal_energy = MISTAKES_WERE_MADE
+	damaged_sm.internal_energy = 0
 	for(var/obj/machinery/power/energy_accumulator/tesla_coil/zappy_boi in range(3, src))
-		zappy_boi.stored_energy = MISTAKES_WERE_MADE
+		zappy_boi.stored_energy = 0
 
 /obj/machinery/atmospherics/components/unary/delam_scram/New()
```

Comment:
I know you didn't touch this proc header but can you change it to be Initialize() instead cuz it absolutely shouldn't be New

Reply:
yes